includes/Handlers/Connection.php

```diff
@@ -272,20 +290,24 @@ public function handle_connect() {
 				throw new SV_WC_API_Exception( 'Invalid nonce' );
 			}
 
-			$merchant_access_token = ! empty( $_GET['merchant_access_token'] ) ? sanitize_text_field( $_GET['merchant_access_token'] ) : '';
+			$is_error                 = ! empty( $_GET['err'] ) ? true : false;
+			$error_code               = ! empty( $_GET['err_code'] ) ? sanitize_text_field( $_GET['err_code'] ) : '';
+			$merchant_access_token    = ! empty( $_GET['merchant_access_token'] ) ? sanitize_text_field( $_GET['merchant_access_token'] ) : '';
+			$system_user_access_token = ! empty( $_GET['system_user_access_token'] ) ? sanitize_text_field( $_GET['system_user_access_token'] ) : '';
+			$system_user_id           = ! empty( $_GET['system_user_id'] ) ? sanitize_text_field( $_GET['system_user_id'] ) : '';
+
+			if ( $is_error && $proxy_error_message = $this->get_message_for_error_code( $error_code ) ) {
+				throw new SV_WC_API_Exception( $proxy_error_message );
```

Comment:
Ah I see - you want to pass a sentence/english version of the error to `SV_WC_API_Exception` constructor.

I'd suggest just adding a new exception type which handles the logging and all the logic related to these errors (using other exception classes as a base if useful). If we need to map to UI text/sentence form in future, could add this behaviour to exception class. 

So the interface would be something like:

```php
if ( $is_error && $error_code ) {
   throw new FacebookApiException( $error_code );
}